### PR TITLE
add 2FA by email mutation

### PR DESCRIPTION
**Pull request by Meyclem**
This PR adds the TwoFA email mutation.

I manually tried it in account and it's working.

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-hh99p2 [connect-js-management] - Add new mutation for email 2FA in sendTwoFAVerificationCode](https://app.clickup.com/t/hh99p2)

File: management/src/commands/send-two-fa-verification-code.ts [hunk 1]
```diff
@@ -57,40 +83,104 @@ async function sendTwoFAVerificationCode(
...
+  managementCredentials: ManagementCredentials,
+  {
+    callbackUrl,
+    identity,
+    userId,
+    localeCodeOverride,
+  }: SendTwoFAVerificationCodeInput,
+): Promise<SendTwoFAVerificationCodeResult> {
+  const twoFAfunction =
+    identity.type.toLocaleLowerCase() === "phone"
```
Comment (Meyclem) [inline management/src/commands/send-two-fa-verification-code.ts:165]: I use `toLocaleLowerCase` to be sure that every casing will match.

**Comment by Oluane** [comment]
Just a little thing, can you update the `README` section concerning `sendTwoFAVerificationCode` please ? 🙏

**Comment by Meyclem** [comment]
> Just a little thing, can you update the `README` section concerning `sendTwoFAVerificationCode` please ? 🙏

I can totally do that

File: management/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.6.1] - 2021-03-15
+
```
Comment (Fuzznimp) [inline management/CHANGELOG.md:6]:
```suggestion

## [0.6.4] - 2021-05-03

```

**Review by Fuzznimp — COMMENTED** [review]
So they discarded their wish to only use the Identity value instead of the Identity object? 🤔

**Comment by Meyclem** [comment]
> So they discarded their wish to only use the Identity value instead of the Identity object? 🤔

I guess so ¯\_(ツ)_/¯

**Comment by Meyclem** [comment]
> Just a little thing, can you update the `README` section concerning `sendTwoFAVerificationCode` please ? 🙏

@Oluane I don't know what to change in the readme.
I didn't change the public interface and it was already describing a code sent by email:

```ts
const input = {
  callbackUrl: "/",
  identity: {
    id: "12488dfe-8e46-4391-a8bb-f0db41078942",
    type: "EMAIL",
    value: "foo@fewlines.co",
    status: "validated",
    primary: true,
  },
  userId: "37b21863-3f38-4d20-848d-3108337a0b8b",
};
```

File: management/src/commands/send-two-fa-verification-code.ts [hunk 3]
```diff
@@ -161,10 +161,16 @@ async function sendTwoFAVerificationCode(
...
-      ? sendPhoneTwoFAVerificationCode
-      : sendEmailTwoFAVerificationCode;
+  const twoFAfunction = ((): TwoFAFunction => {
+    switch (identity.type.toLocaleLowerCase()) {
+      case "phone":
+        return sendPhoneTwoFAVerificationCode;
+      case "email":
+        return sendEmailTwoFAVerificationCode;
+      default:
+        throw new InvalidIdentityTypeError();
```
Comment (Meyclem) [inline management/src/commands/send-two-fa-verification-code.ts:171]: This way we are sure that only `email` and `phone` identity types are ok. All the other would throw `InvalidIdentityTypeError`

**Comment by Oluane** [comment]
> @Oluane I don't know what to change in the readme.
> I didn't change the public interface and it was already describing a code sent by email:

Yes, I still had in mind the changes of identity input to just identity value. My bad, forget about it 🙂

**Review by Oluane — APPROVED** [review]

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**